# HG changeset patch
# User Rik <rik@octave.org>
# Date 1482768484 28800
#      Mon Dec 26 08:08:04 2016 -0800
# Branch stable
# Node ID 2b2ac017c901a481daa1fae49ad6d30c4c37a95a
# Parent  06513f1f90afeb1bab38698dff5d10726240c9ab
Fix undefined return argument for more than 2 outputs from ode solver (bug #49890).

* ode23.m, ode45.m: Change test for "nargout == 5" to "nargout > 2".

diff --git a/scripts/ode/ode23.m b/scripts/ode/ode23.m
--- a/scripts/ode/ode23.m
+++ b/scripts/ode/ode23.m
@@ -274,17 +274,17 @@ function varargout = ode23 (fun, trange,
       varargout{1}.stats = struct ();
       varargout{1}.stats.nsteps   = nsteps;
       varargout{1}.stats.nfailed  = nfailed;
       varargout{1}.stats.nfevals  = nfevals;
       varargout{1}.stats.npds     = npds;
       varargout{1}.stats.ndecomps = ndecomps;
       varargout{1}.stats.nlinsols = nlinsols;
     endif
-  elseif (nargout == 5)
+  elseif (nargout > 2)
     varargout = cell (1,5);
     varargout{1} = solution.t;
     varargout{2} = solution.x;
     if (! isempty (odeopts.Events))
       varargout{3} = solution.event{3};  # Time info when an event occurred
       varargout{4} = solution.event{4};  # Results when an event occurred
       varargout{5} = solution.event{2};  # Index info which event occurred
     endif
diff --git a/scripts/ode/ode45.m b/scripts/ode/ode45.m
--- a/scripts/ode/ode45.m
+++ b/scripts/ode/ode45.m
@@ -268,17 +268,17 @@ function varargout = ode45 (fun, trange,
       varargout{1}.stats = struct ();
       varargout{1}.stats.nsteps   = nsteps;
       varargout{1}.stats.nfailed  = nfailed;
       varargout{1}.stats.nfevals  = nfevals;
       varargout{1}.stats.npds     = npds;
       varargout{1}.stats.ndecomps = ndecomps;
       varargout{1}.stats.nlinsols = nlinsols;
     endif
-  elseif (nargout == 5)
+  elseif (nargout > 2)
     varargout = cell (1,5);
     varargout{1} = solution.t;
     varargout{2} = solution.x;
     if (! isempty (odeopts.Events))
       varargout{3} = solution.event{3};  # Time info when an event occurred
       varargout{4} = solution.event{4};  # Results when an event occurred
       varargout{5} = solution.event{2};  # Index info which event occurred
     endif
